File: java/clock/src/main/java/Clock.java
class Clock {
  private static final int MINUTES_IN_DAY = 24 * 60;
  private int minutes;

  Clock(int h, int m) {
    minutes = normalizeMinutes(h * 60 + m);
  }

  void add(int m) {
    minutes = normalizeMinutes(minutes + m);
  }

  @Override
  public String toString() {
    return String.format("%02d:%02d", minutes / 60, minutes % 60);
  }

  @Override
  public boolean equals(Object clock) {
    return minutes == ((Clock)clock).minutes;
  }

  private int normalizeMinutes(int m) {
    return ((m % MINUTES_IN_DAY) + MINUTES_IN_DAY) % MINUTES_IN_DAY;
  }
}

File: csharp/clock/Clock.cs
public class Clock
{
    int minutes;
    private const int MinutesInDay = 60 * 24;

    public Clock(int hours, int minutes)
    {
        this.minutes = ((((hours * 60) + minutes) % MinutesInDay) + MinutesInDay) % MinutesInDay;
    }

    public Clock Add(int minutesToAdd) => new Clock(0, minutes + minutesToAdd);

    public Clock Subtract(int minutesToSubtract) => Add(-minutesToSubtract);
    
    public override string ToString() => $"{(minutes / 60):D2}:{(minutes % 60):D2}";

    public override bool Equals(object obj) => this.ToString() == obj.ToString();

    public override int GetHashCode() => this.ToString().GetHashCode();
}
